Functional testing example, Loc.is_open_for_business, mobility changes and max queue size

markovg opened the pull request:
# Description


Addressing comments and discussion around commit 94181da
https://github.com/markovg/covid_p2p_simulation/commit/94181da8ffde60c625cae27680dffa3ba6b133a4
- Moved queue length checks to _select_location, added for misc
- rename is_open -> is_open_for_business
- remove self.dead attribute, make self.is_dead a property
  that queries truth value of self.recovered_timestamp == datetime.datetime.max

commit 94181da
- Implemented some significant model changes to mobility
- Implemented a first simple functional test of an infection at a seniors residence
- Added a store maximum queue size
  - added filter_queue_max as a util to prevent humans joining long queues
- Added Location.is_open_for_business to implement lock-down
- Added SYMPTOMS_META_IDMAP, and lazy City.hd to reduce some entropy
- Added EmptyCity to manually build simple experiments for functional testing
- added Human.expire(self) to reduce entropy surrounding various death code paths
- Fixed a bug that dead humans were not counted as is_removed
- Several TODO comments of things that need looking into
- Added filter_open as a util to compute mobility with lock-down Locations
- Added logic for working from home when workplace has lockdown
- Retired people now don't go to work
- fixed cold->flu typo/bug

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Added and ran a new funtional test.  

- [x] tests/test_functional_seniors_residence.py

markovg committed:
- Implemented some significant model changes to mobility
- Implemented a first simple functional test of an infection at a seniors residence
- Added a store maximum queue size
- Added Location.is_open to implement lockdown
- Added SYMPTOMS_META_IDMAP, and lazy City.hd to reduce some entropy
- Added ExternalBuildCity to manually build simple experiments for functional testing
- added Human.expire(self) to reduce entropy surounding various death code paths
- Fixed a bug that dead humans were not counted as is_removed
- Several TODO comments of things that need looking into
- Added filter_open as a util to compute mobility with lockeddown Locations
- Added logic for working from home when workplace has lockdown
- Retired people now don't go to work
- fixed cold->flu typo/bug
- 1st step to fixing p_infection n_infectious_contacts hack, more to follow

markovg committed:
Addressing comments and discussion around commit 94181da
https://github.com/markovg/covid_p2p_simulation/commit/94181da8ffde60c625cae27680dffa3ba6b133a4
- Moved queue length checks to _select_location, added for misc
- rename is_open -> is_open_for_business
- remove self.dead attribute, make self.is_dead a property
that queries truth value of self.recovered_timestamp == datetime.datetime.max

markovg committed:
Fix to make tests/models_test.py pass

markovg committed:
Merge branch 'experimental' into experimental

pg2455 commented on src/covid19sim/simulator.py:
```diff
@@ -1383,7 +1382,8 @@ def _select_location(self, location_type, city):
             pool_locs = [m for m in city.miscs if m != self.location]
-            locs = filter_open(city.miscs)
+            # Only consider locations open for business and not too long queues
+            locs = filter_open_and_short_queue(city.miscs)
```
shorten the names? `filter_open` and `filter_queue_max`.

pg2455 commented on src/covid19sim/frozen/helper.py:
```diff
@@ -47,6 +47,10 @@
 
+# Index SYMPTOMS_META by ID
+SYMPTOMS_META_IDMAP = [""] * len(SYMPTOMS_META)
+for k, v in SYMPTOMS_META.items():
```
@satyaog  - Can you have a look at this?

pg2455 reviewed (commented):
Great work! Thanks a lot 👍

markovg committed:
Implemented discussed chages for PR #144
https://github.com/pg2455/covid_p2p_simulation/pull/144
Specifically, the changes are:
- Fix for cached_property py<3.8 - conditional import & requirements.txt
- Revert p_infection change ( removal of '* h.n_infectious_contacts')
  - Thats for future commit/PR
- Revised doc strings for is_dead, filter_*, expire, EmptyCity.initWorld
- Revised filter_* -> filter_open, filter_queue_max
- Use pathlib for accessing functional configs
- Rename directory functional_configs -> test_configs
- ExternalBuildCity renamed to EmptyCity and moved to base.py
- remove filter_open!=[] in Human.run mobility block (its handled by Human.excursion)

markovg committed:
Merge branch 'experimental' of github.com:markovg/covid_p2p_simulation into experimental

pg2455 reviewed (approved):
Thanks!
